libparsec/crates/client/tests/unit/client/update_user_profile.rs

```diff
@@ -0,0 +1,109 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use libparsec_client_connection::test_register_send_hook;
+use libparsec_protocol::authenticated_cmds;
+use libparsec_tests_fixtures::prelude::*;
+use libparsec_types::prelude::*;
+
+use crate::ClientUserUpdateProfileError;
+
+use super::utils::client_factory;
+
+#[parsec_test(testbed = "coolorg", with_server)]
+async fn ok(env: &TestbedEnv) {
+    let alice = env.local_device("alice@dev1");
+    let client = client_factory(&env.discriminant_dir, alice).await;
+
+    let bob_user_id: UserID = "bob".parse().unwrap();
+
+    let users = client.list_users(false, None, None).await.unwrap();
+    let bob = users.iter().find(|user| user.id == bob_user_id).unwrap();
+    p_assert_eq!(bob.current_profile, UserProfile::Standard);
+
+    client
+        .update_user_profile(bob_user_id, UserProfile::Admin)
+        .await
+        .unwrap();
+
+    let users = client.list_users(false, None, None).await.unwrap();
+    let bob = users.iter().find(|user| user.id == bob_user_id).unwrap();
+    p_assert_eq!(bob.current_profile, UserProfile::Admin);
+
+    client
+        .update_user_profile(bob_user_id, UserProfile::Outsider)
+        .await
+        .unwrap();
+
+    let users = client.list_users(false, None, None).await.unwrap();
+    let bob = users.iter().find(|user| user.id == bob_user_id).unwrap();
+    p_assert_eq!(bob.current_profile, UserProfile::Outsider);
+}
+
+#[parsec_test(testbed = "coolorg", with_server)]
+async fn not_allowed(env: &TestbedEnv) {
```

Comment:
This tests that a user with Standard profile cannot update other user's profile.

We should test the same thing for a user with Outsider profile.